From: Fernando Fernandez Mancera <fmancera@suse.de>
Date: Mon Sep 8 16:58:11 2025 +0200
Subject: kABI: netfilter flowtable move gc operation to bottom
References: git-fixes
Patch-mainline: Never, kABI workaround

Move the gc pointer to function to the bottom of the struct definition. There
is only a hole available but it is 4 bytes, so not enough.

Signed-off-by: Fernando Fernandez Mancera <fmancera@suse.de>

---
 include/net/netfilter/nf_flow_table.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/net/netfilter/nf_flow_table.h
+++ b/include/net/netfilter/nf_flow_table.h
@@ -53,7 +53,6 @@ struct nf_flowtable_type {
 	struct list_head		list;
 	int				family;
 	int				(*init)(struct nf_flowtable *ft);
-	bool				(*gc)(const struct flow_offload *flow);
 	int				(*setup)(struct nf_flowtable *ft,
 						 struct net_device *dev,
 						 enum flow_block_command cmd);
@@ -64,6 +63,9 @@ struct nf_flowtable_type {
 	void				(*free)(struct nf_flowtable *ft);
 	nf_hookfn			*hook;
 	struct module			*owner;
+#ifndef __GENKSYMS__
+	bool				(*gc)(const struct flow_offload *flow);
+#endif
 };
 
 enum nf_flowtable_flags {
